### DISCUSSION TITLE
Ability to insert a date/time stamp

### DISCUSSION THREAD

**Discussion by nathanl** [damionx7/Zettel-Notes-Documentation#136] [General]
First: thanks for Zettel, I love it! :) Especially being able to sync with Git and edit the same files on my laptop.

For some kinds of notes, I like to keep a running log in a single file, where I make dates tamped entries.

Currently, I have to type in the date myself when using Zettel, and I often make typos. If I edit the same file from Vim, I have a shortcut to run `date -j +"## %Y-%m-%d (%a)"`, which outputs something like `## 2024-07-26 (Fri)`.

So I'd like a way to insert date/time stamps in a user-specified format like the one above.

**Comment by damionx7** [comment]
Hey! Have you looked at Log Notes https://znotes.thedoc.eu.org/note/note-types/?h=log+note#log-note and Text Snippets https://znotes.thedoc.eu.org/note/editor/?h=snippe#text-snippets-button

**Reply by damionx7** [reply]
You can also use custom date format in text snippets according to guidelines at https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html.

**Reply by nathanl** [reply]
Ha, there are lots of features I didn't know about! Also I never knew I could scroll that row of buttons where Text Snippets is. 😆  (Now I see that I can also rearrange their order if I go to the end and click the gear. Awesome!)  I think Text Snippets would be preferable for me. How do I indicate that my `printf` string components, like `%Y` or whatever, should be expanded? So far they're being output as-is, not replaced with the current date's info.

**Reply by damionx7** [reply]
:laughing: . You can surround the date format with `$$` eg. ## `$$yyyy-MM-hh$$`.

**Reply by nathanl** [reply]
Ah, OK, it appears it's using `SimpleDateFormat`, not `strftime` format. `## $$yyy-MM-hh (EE)$$` gets me `## 2024-07-26 (Fri)` as in my example above. Thanks!

**Reply by nathanl** [reply]
Thanks for your help!  I'll close this discussion. I documented what you taught me in https://github.com/damionx7/Zettel-Notes-Documentation/pull/138
